include/rmm/mr/device/pool_memory_resource.hpp

```diff
@@ -87,34 +87,27 @@ class pool_memory_resource final
    * the available memory on the current device.
    */
   explicit pool_memory_resource(Upstream* upstream_mr,
-                                std::size_t initial_pool_size = default_initial_size,
-                                std::size_t maximum_pool_size = default_maximum_size)
+                                thrust::optional<std::size_t> initial_pool_size = thrust::nullopt,
+                                thrust::optional<std::size_t> maximum_pool_size = thrust::nullopt)
     : upstream_mr_{[upstream_mr]() {
         RMM_EXPECTS(nullptr != upstream_mr, "Unexpected null upstream pointer.");
         return upstream_mr;
       }()}
   {
-    RMM_EXPECTS(
-      initial_pool_size == default_initial_size ||
-        (initial_pool_size == rmm::detail::align_up(initial_pool_size, allocation_alignment)),
-      "Error, Initial pool size required to be a multiple of 256 bytes");
-    RMM_EXPECTS(
-      maximum_pool_size == default_maximum_size ||
-        (maximum_pool_size == rmm::detail::align_up(maximum_pool_size, allocation_alignment)),
-      "Error, Maximum pool size required to be a multiple of 256 bytes");
+    RMM_EXPECTS(rmm::detail::is_aligned(initial_pool_size.value_or(0), allocation_alignment),
+                "Error, Initial pool size required to be a multiple of 256 bytes");
+    RMM_EXPECTS(rmm::detail::is_aligned(maximum_pool_size.value_or(0), allocation_alignment),
+                "Error, Maximum pool size required to be a multiple of 256 bytes");
 
     std::size_t free{}, total{};
     std::tie(free, total) = detail::available_device_memory();
 
-    initial_pool_size = (initial_pool_size == default_initial_size)
-                          ? rmm::detail::align_up(std::min(free, total / 2), allocation_alignment)
-                          : initial_pool_size;
-    current_pool_size_ = initial_pool_size;
-    maximum_pool_size_ = (maximum_pool_size == default_maximum_size)
-                           ? rmm::detail::align_down(free, allocation_alignment)
-                           : maximum_pool_size;
+    current_pool_size_ = initial_pool_size.value_or(
```

Comment:
Is `initial_pool_size` guaranteed to be aligned? (does it matter?)

Reply:
Keep reading...